torch/hub.py

```diff
@@ -602,9 +603,18 @@ def download_url_to_file(url, dst, hash_prefix=None, progress=True):
     # We deliberately save it in a temp file and move it after
     # download is complete. This prevents a local working checkpoint
     # being overridden by a broken download.
+    # We deliberately do not use NamedTemporaryFile to avoid restrictive
```

Comment:
What about using `NamedTemporaryFile` and then chmod it after? Does that work?

Reply:
@dagitses It's possible, but seems to be more risky and hard to get right. For reference you can see that BorgBackup originally took this approach: https://github.com/borgbackup/borg/commit/0f9d88bf2e1f2d14be8fa30810948ef0e4107f42 (but was recently changed, see below). My two issues with using `NamedTemporaryFile` and changing permissions afterwards are:

1. There is no API for querying umask, so you need to set the umask (which returns the old value), then set it back. This could introduce thread safety issues (i.e. a file is created in between these two instructions).
2. Something could be "left out" when simply using chmod. This was very recently discovered in BorgBackup (https://github.com/borgbackup/borg/issues/6933) and resulted in them switching to taking the approach I am here (custom temp file creation). I'd also be worried about the behaviour on different operating systems or in some other edge case.